daemon/containerd/progress.go

```diff
@@ -54,12 +53,9 @@ func showProgress(ctx context.Context, ongoing *jobs, w io.Writer, updateFunc up
 					logrus.WithError(err).Error("Updating progress failed")
 					return
 				}
-
-				if done {
-					return
-				}
 			case <-ctx.Done():
-				done = true
+				updateFunc(ctx, ongoing, out, start)
```

Comment:
One last question (before I blame myself for not asking later); do we still need to check `ongoing.IsResolved()` here? Because we don't know "when" we reach this code. Or would calling `updateFunc()` be fine if `ongoing` is not in that state?

```suggestion
				if ongoing.IsResolved() {
					// update progress once more on cancellation or timeout
					updateFunc(ctx, ongoing, out, start)
				}
```

Reply:
Generally calling the updateFunc is fine even if `IsResolved` is false. With the current push/pull progress handlers this is useless anyway, as they just loop over all jobs.
The IsResolved was extracted from the initial pull progress code, so I don't really know if I understand the intent behind it correctly, but to me it's just a safeguard before updating the progress when the jobs hasn't been added yet. 
If the progress ends completely, then calling updateFunc without any jobs is still useful IMO, because it may add some kind of "Finished" status update or information that there weren't anything to be done. But this isn't something that is really done by the push/pull progress code we have.

Reply:
Thanks for checking! Just wanted to be sure we didn't miss something (and, e.g. get a `panic` because some data wasn't set 😅)